addons/knowledge/models/knowledge_article.py

```diff
@@ -276,6 +286,12 @@ def _search_is_user_favourite(self, operator, value):
         else:
             return [('favourite_user_ids', 'not in', [self.env.user.id])]
 
+    def _search_main_article_id(self, operator, value):
```

Comment:
As I said, even if recursive search works (not even sure of it), this would maybe not performant at all.
I would go for : 
1. limit operator to '=' (else NotImplementedError)
2. retreive article based on id or name (like in _search_owner_id)
3. retreive all the children of that article (and all the children of the children etc..)
4. return id in descendants.ids